pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerTest.java

```diff
@@ -816,6 +819,48 @@ public void testSegmentReplacementForRefresh()
     Assert.assertEquals(new HashSet<>(ControllerTestUtils.getHelixResourceManager()
             .getSegmentsFor(OFFLINE_SEGMENTS_REPLACE_TEST_REFRESH_TABLE_NAME, true)),
         new HashSet<>(Arrays.asList("s9", "s10", "s11")));
+
+    // Call revert segment replacements
+    ControllerTestUtils.getHelixResourceManager()
+        .revertReplaceSegments(OFFLINE_SEGMENTS_REPLACE_TEST_REFRESH_TABLE_NAME, lineageEntryId3, false);
+    waitForSegmentsToDelete(OFFLINE_SEGMENTS_REPLACE_TEST_REFRESH_TABLE_NAME, 3, MAX_TIMEOUT_IN_MILLISECOND);
+    Assert.assertEquals(new HashSet<>(ControllerTestUtils.getHelixResourceManager()
+            .getSegmentsFor(OFFLINE_SEGMENTS_REPLACE_TEST_REFRESH_TABLE_NAME, true)),
+        new HashSet<>(Arrays.asList("s3", "s4", "s5")));
+
+    // Re-upload (s9, s10, s11) to test the segment clean up from startReplaceSegments
```

Comment:
What's the purpose of this step? In which circumstance will we upload the original segments again?

Reply:
After the code change, we delete `segmentsTo` in 2 different path.

1. `revertReplaceSegments` API will delete `segmentsTo`
2. `startReplaceSegments` API will also try to clean up `segmentsTo` for `REVERTED` lineage. 

I tested the scenario 1. Now, I need to test the scenario 2. So, I am just trying to set up the testing condition because revert call in the previous line deleted `s9, s10, s11`.

Reply:
Thanks for the explanation! Could you update the comment to the code as well? I guess ppl will also have the same question on it. You can also elaborate on L848 that (s9, s10, s11) will be removed after making the end call.

Reply:
added more comments.